redundant-bmc/src/rbmc_tool.cpp

```diff
@@ -0,0 +1,174 @@
+// SPDX-License-Identifier: Apache-2.0
+
+#include "services_impl.hpp"
+
+#include <CLI/CLI.hpp>
+#include <xyz/openbmc_project/State/BMC/Redundancy/Sibling/client.hpp>
+#include <xyz/openbmc_project/State/BMC/Redundancy/client.hpp>
+#include <xyz/openbmc_project/State/BMC/client.hpp>
+
+#include <format>
+#include <iostream>
+
+using Sibling =
+    sdbusplus::client::xyz::openbmc_project::state::bmc::redundancy::Sibling<>;
+using Redundancy =
+    sdbusplus::client::xyz::openbmc_project::state::bmc::Redundancy<>;
+using BMCState = sdbusplus::client::xyz::openbmc_project::state::BMC<>;
+using Role = Redundancy::Role;
+
+// NOLINTBEGIN
+sdbusplus::async::task<>
+    displayLocalBMCInfo(sdbusplus::async::context& ctx, bool extended)
+// NOLINTEND
+{
+    auto redundancy = sdbusplus::async::proxy()
+                          .service(Redundancy::interface)
+                          .path(Redundancy::instance_path)
+                          .interface(Redundancy::interface);
+
+    std::cout << "Local BMC\n";
+    std::cout << "-----------------------------\n";
+
+    try
+    {
+        auto props =
+            co_await redundancy
+                .get_all_properties<Redundancy::PropertiesVariant>(ctx);
+
+        auto role =
+            Redundancy::convertRoleToString(std::get<Role>(props.at("Role")));
+        role = role.substr(role.find_last_of('.') + 1);
+        std::cout << std::format("Role:                {}\n", role);
+
+        rbmc::ServicesImpl services{ctx};
+        std::cout << std::format("BMC Position:        {}\n",
+                                 services.getBMCPosition());
+
+        auto enabled = std::get<bool>(props.at("RedundancyEnabled"));
+        std::cout << std::format("Redundancy Enabled:  {}\n", enabled);
+
+        if (extended)
```

Comment:
Do you think adding all CFAM fields for the local BMC, similar to how it was done for the sibling BMC, would be helpful for the BMC dump?

Reply:
We can get the actual CFAM fields for the dump by using cfamstool, which has them for the local BMC (and sibling).

I guess here the local BMC doesn't have the BMC state and SiblingComm fields.  I didn't add the BMC state since the code had no use for it yet, but yea I might as well add it and the other one so it has them all.

Reply:
Added BMC state to the output.  Let me know if you were meaning something else.

Reply:
I thought dumping all these [fields](https://github.com/ibm-openbmc/openpower-proc-control/tree/1110/rbmc-cfam-daemon#cfam-s-scratchpad-register-layout) for both BMCs would be helpful for debugging. However, we can always add it later if needed.